packages/grid/_modules_/grid/components/cell/GridDetailPanelToggleCell.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { unstable_composeClasses as composeClasses } from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
+import { GridRenderCellParams } from '../../models/params/gridCellParams';
+import { getDataGridUtilityClass } from '../../gridClasses';
+import { GridComponentProps } from '../../GridComponentProps';
+import { gridExpandedRowsContentCache } from '../../hooks/features/detailPanel/gridDetailPanelSelector';
+import { useGridSelector } from '../../hooks/utils/useGridSelector';
+
+type OwnerState = { classes: GridComponentProps['classes']; isExpanded: boolean };
+
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { classes, isExpanded } = ownerState;
+
+  const slots = {
+    root: ['detailPanelToggleCell', isExpanded && 'detailPanelToggleCell--expanded'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+const GridDetailPanelToggleCell = (props: GridRenderCellParams) => {
+  const { id, value: isExpanded } = props;
+
+  const rootProps = useGridRootProps();
+  const apiRef = useGridApiContext();
+  const ownerState: OwnerState = { classes: rootProps.classes, isExpanded };
+  const classes = useUtilityClasses(ownerState);
+
+  const contentCache = useGridSelector(apiRef, gridExpandedRowsContentCache);
```

Comment:
Do you mean to add a `hasDetailPanel(row)` method to the API? Similar to `isColumnPinned(field)`.

Reply:
Not exactly. For now, developers have access to two parameters:
- `getDetailPanelContent`
- `getDetailPanelHeight`

And if you do not want to have a detail panel for some row, the `getDetailPanelContent` must return `null` or `undefined`. With this method, you have to evaluate all the `getDetailPanelContent(row)` to know if the detail panel is a valid component. I'm proposing to let developers provide a third prop `disableDetailPannel(row)`.

I thought about it for two reasons:
- Using `React.isValidElement` seems a bit hacky.
- I imagine for developers, knowing if the row has a detail panel will be easy. It will probably be a condition such as: "the row status is open"